main.go

```diff
@@ -257,16 +261,29 @@ func main() {
 	ctrl.SetLogger(klogr.New())
 
 	if enableTracing {
-		flush, err := initJaegerTracing()
+		tp, err := jaegerTracerProvider("http://localhost:5778/api/traces")
 		if err != nil {
 			setupLog.Error(err, "failed to init Jaeger tracing")
 			os.Exit(1)
 		}
+		// Register our TracerProvider as the global so any imported
+		// instrumentation in the future will default to using it.
+		otel.SetTracerProvider(tp)
 
 		tracing.Register(ot.NewOpenTelemetryAutorestTracer(tele.Tracer()))
 
-		// try to flush all traces before exiting
-		defer flush()
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		// Cleanly shutdown and flush telemetry when the application exits.
+		defer func(ctx context.Context) {
+			// Do not make the application hang when it is shutdown.
+			ctx, cancel = context.WithTimeout(ctx, time.Second*5)
+			defer cancel()
+			if err := tp.Shutdown(ctx); err != nil {
+				setupLog.Error(err, "failed to shut down Jaeger tracing")
+			}
+		}(ctx)
```

Comment:
The use of `ctx` here is ineffectual. You might as well pass in `context.Background`. 

Please consider setting up something like [Cluster API's signal handler](https://github.com/kubernetes-sigs/cluster-api/blob/19b900defb84434e3a23d786c311c5daf9d94d6d/main.go#L200-L201) to gracefully stop and flush the telemetry when the process receives OS signals to terminate.

Reply:
Good call--I'll check out the signal handler.

Reply:
I changed the naive code (copied from otel examples, where it's also ineffectual...) to instead use the context set up by the signal handler.